src/main/java/com/google/firebase/projectmanagement/FirebaseProjectManagementServiceImpl.java

```diff
@@ -0,0 +1,909 @@
+    this.requestFactory =
+        httpTransport.createRequestFactory(new FirebaseRequestInitializer(app));
+
+    this.scheduledExecutor = new FirebaseScheduledExecutor(
+        ImplFirebaseTrampolines.getThreadFactory(app), "firebase-project-management");
+  }
+
+  @VisibleForTesting
+  void setInterceptor(HttpResponseInterceptor interceptor) {
+    this.interceptor = interceptor;
+  }
+
+  void destroy() {
+    // NOTE: Any instance of IosApp, AndroidApp, or FirebaseProjectManagement that depends on this
+    // instance will no longer be able to make RPC calls. All polling or waiting iOS or Android App
+    // creations will be interrupted, even though the initial creation RPC (if made successfully) is
+    // still processed normally (asynchronously) by the server.
+    scheduledExecutor.shutdownNow();
+  }
+
+  /* getAndroidApp */
+
+  @Override
+  public AndroidAppMetadata getAndroidApp(String appId) throws FirebaseProjectManagementException {
+    return getAndroidAppOp(appId).call();
+  }
+
+  @Override
+  public ApiFuture<AndroidAppMetadata> getAndroidAppAsync(String appId) {
+    return getAndroidAppOp(appId).callAsync(app);
+  }
+
+  private CallableOperation<AndroidAppMetadata, FirebaseProjectManagementException> getAndroidAppOp(
+      final String appId) {
+    return new CallableOperation<AndroidAppMetadata, FirebaseProjectManagementException>() {
+      @Override
+      protected AndroidAppMetadata execute() throws FirebaseProjectManagementException {
+        String url = String.format(
+            "%s/v1beta1/projects/-/androidApps/%s", FIREBASE_PROJECT_MANAGEMENT_URL, appId);
+        AndroidAppResponse parsedResponse = new AndroidAppResponse();
+        makeGetRequest(url, parsedResponse, appId, "App ID");
+        return new AndroidAppMetadata(
+            parsedResponse.name,
+            parsedResponse.appId,
+            Strings.nullToEmpty(parsedResponse.displayName),
+            parsedResponse.projectId,
+            parsedResponse.packageName);
+      }
+    };
+  }
+
+  /* getIosApp */
+
+  @Override
+  public IosAppMetadata getIosApp(String appId) throws FirebaseProjectManagementException {
+    return getIosAppOp(appId).call();
+  }
+
+  @Override
+  public ApiFuture<IosAppMetadata> getIosAppAsync(String appId) {
+    return getIosAppOp(appId).callAsync(app);
+  }
+
+  private CallableOperation<IosAppMetadata, FirebaseProjectManagementException> getIosAppOp(
+      final String appId) {
+    return new CallableOperation<IosAppMetadata, FirebaseProjectManagementException>() {
+      @Override
+      protected IosAppMetadata execute() throws FirebaseProjectManagementException {
+        String url = String.format(
+            "%s/v1beta1/projects/-/iosApps/%s", FIREBASE_PROJECT_MANAGEMENT_URL, appId);
+        IosAppResponse parsedResponse = new IosAppResponse();
+        makeGetRequest(url, parsedResponse, appId, "App ID");
+        return new IosAppMetadata(
```

Comment:
Same here. Provide a constructor that takes IosAppResponse as the only argument.